src/univariate/continuous/pgeneralizedgaussian.jl

```diff
@@ -0,0 +1,110 @@
+
+"""
+    PGeneralizedGaussian(α, μ, p)
+
+The *p-Generalized Gaussian distribution*, more commonly known as the exponential
+power or the generalized normal distribution, with scale 'α', location 'μ', and
+shape 'p' has the probability density function
+
+```math
+f(x, \\mu, \\alpha, p) = \\frac{p}{2\\alpha\\Gamma(1/p)} e^{-(\\frac{|x-\\mu|}{\\alpha})^p} \\quad x \\in (-\\infty, +\\infty) , \\alpha > 0, p > 0
+```
+
+The Generalized Gaussian (GGD) is a parametric distribution that incorporates the
+Normal and Laplacian distributions as special cases where p = 1 and p = 2. As p → ∞,
+the distribution approaches the Uniform distribution on (μ-α, μ+α).
+
+```julia
+PGeneralizedGaussian()           # GGD with shape 2, scale 1, location 0, (the Normal distribution)
+PGeneralizedGaussian(loc,s,sh)   # GGD with location loc, scale s, and shape sh
+
+params(d)                       # Get the parameters, i.e. (loc,s,sh,)
+shape(d)                        # Get the shape parameter, sh
+scale(d)                        # Get the scale parameter, s
+location(d)                     # Get the location parameter, loc
+```
+
+External Links
+ * [Generalized Gaussian on Wikipedia](http://en.wikipedia.org/wiki/Generalized_normal_distribution)
+ * [R Package reference](https://www.researchgate.net/publication/254282790_Simulation_of_the_p-generalized_Gaussian_distribution)
+ """
+
+struct PGeneralizedGaussian{T<:Real} <: ContinuousUnivariateDistribution
```

Comment:
The three parameters should be allowed to have different number types, for example for auto-differentiation. Can be done with:
```julia
struct PGeneralizedGaussian{T1<:Real, T2<:Real, T31<:Real} <: ContinuousUnivariateDistribution
    α::T1
    μ::T2
    p::T3
     PGeneralizedGaussian(μ::T1,α::T2,p::T3) where {T1,T2,T3} = (@check_args(PGeneralizedGaussian, α > zero(α) && p > zero(p)); new{T1,T2,T3}(μ,α,p)) 
```

Reply:
That sounds good to me, I'm unsure however how that would matriculate down to the other uses of T in the methods - would that remain the same, or would we similarly reformat the other parameter types as well?

Reply:
yup you need to reformat the rest of the methods and constructors

Reply:
Understood - I should be able to get to this later, feel free to make any other changes. Thank you for bearing with me and making this robust!